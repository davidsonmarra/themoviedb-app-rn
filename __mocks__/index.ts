import MovieDTO from '../src/@types/movies-dto';

export const mockCardMovie: MovieDTO = {
  id: 1,
  title: 'Movie Title',
  overview: 'Movie Overview',
  poster_path: '7qOSKoOAPgemYhBwbJgBWcCxPWZ.jpg',
  release_date: '2021-01-01',
  backdrop_path: 'path/to/image',
  vote_average: 10,
  genre_ids: [1, 2],
};

export const mockListMovie: MovieDTO[] = [
  {
    id: 1,
    title: 'Movie Title',
    overview: 'Movie Overview',
    poster_path: 'path/to/image',
    release_date: '2021-01-01',
    backdrop_path: 'path/to/image',
    vote_average: 10,
    genre_ids: [1, 2],
  },
  {
    id: 2,
    title: 'Movie Title 2',
    overview: 'Movie Overview 2',
    poster_path: 'path/to/image',
    release_date: '2021-01-01',
    backdrop_path: 'path/to/image',
    vote_average: 9,
    genre_ids: [1, 2],
  },
];
