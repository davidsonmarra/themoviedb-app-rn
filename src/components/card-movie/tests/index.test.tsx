import React from 'react';
import {render, screen} from '@testing-library/react-native';
import {mockCardMovie} from '../../../../__mocks__';
import CardMovie from '..';

const renderComponent = (props: any) => render(<CardMovie {...props} />);

describe('CardMovie', () => {
  it('renders correctly', () => {
    const tree = renderComponent({movie: mockCardMovie}).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders children correctly', () => {
    renderComponent({movie: mockCardMovie});
    const title = screen.getByText(mockCardMovie.title);
    expect(title.children).toEqual([mockCardMovie.title]);
  });

  it('should call onPress', () => {
    const onPressCard = jest.fn();
    const button = renderComponent({
      movie: mockCardMovie,
      onPressCard,
    }).root;
    button.props.onClick();
    expect(onPressCard).toHaveBeenCalledWith(mockCardMovie);
  });
});
