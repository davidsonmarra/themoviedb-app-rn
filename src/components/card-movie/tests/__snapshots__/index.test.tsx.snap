// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardMovie renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderColor": "#CCCCCC",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "marginBottom": 16,
      "opacity": 1,
    }
  }
>
  <Image
    source={
      {
        "uri": "https://image.tmdb.org/t/p/w188_and_h282_bestv2/7qOSKoOAPgemYhBwbJgBWcCxPWZ.jpg",
      }
    }
    style={
      {
        "borderBottomLeftRadius": 8,
        "borderTopLeftRadius": 8,
        "height": 141,
        "width": 94,
      }
    }
  />
  <View
    style={
      {
        "flex": 1,
        "padding": 12,
      }
    }
  >
    <Text
      style={
        {
          "color": "#121212",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Movie Title
    </Text>
    <Text
      style={
        {
          "color": "#808080",
          "fontSize": 14,
        }
      }
    >
      31/12/2020
    </Text>
    <Text
      numberOfLines={3}
      style={
        {
          "color": "#222222",
          "fontSize": 14,
          "marginTop": 8,
        }
      }
    >
      Movie Overview
    </Text>
  </View>
</View>
`;
